spirit/core/utils/markdown/block.py

```diff
@@ -41,12 +41,22 @@ class BlockGrammar(mistune.BlockGrammar):
     # * https://www.youtube.com/watch?v=Z0UISCEe52Y
     # * http://youtu.be/afyK1HSFfgw
     # * https://www.youtube.com/embed/vsF0K3Ou1v0
+    #
+    # Also works for timestamps:
+    # * https://www.youtube.com/watch?v=Z0UISCEe52Y&t=1m30s
+    # * https://www.youtube.com/watch?v=O1QQajfobPw&t=1h1m38s
+    # * https://www.youtube.com/watch?v=O1QQajfobPw&feature=youtu.be&t=3698
+    # * https://youtu.be/O1QQajfobPw?t=3698
+    # * https://youtu.be/O1QQajfobPw?t=1h1m38s
+    #
     youtube = re.compile(
         r'^https?://(www\.)?'
         r'(youtube\.com/watch\?v='
         r'|youtu\.be/'
         r'|youtube\.com/embed/)'
         r'(?P<id>[a-zA-Z0-9_\-]{11})'
+        r'((&|\?)?feature=youtu.be)?'
+        r'((&|\?)?t=(?P<start_hours>[0-9]+h)?(?P<start_minutes>[0-9]+m)?(?P<start_seconds>[0-9]+s?)?)?'
```

Comment:
The regex won't match when the parameters are not in that specific order. I think there are two ways to fix it, match all params and then parse them later or this:

```python
youtube = re.compile(
        r'^https?://(www\.)?'
        r'(youtube\.com/watch\?v='
        r'|youtu\.be/'
        r'|youtube\.com/embed/)'
        r'(?P<id>[a-zA-Z0-9_\-]{11})'
        r'((&|\?)('
        r'|(t=(?P<start_hours>[0-9]{1,2}h)?(?P<start_minutes>[0-9]{1,4}m)?(?P<start_seconds>[0-9]{1,5}s?)?)'
        r'|([^&\s]+)'
        r')){,10}'
        r'(?:\n+|$)'
    )
```

It's hard to read, but what it does is match up to 10 parameters, it only cares about `t=`, all the other params are skipped. It also does not care about the parameters order. When the same parameter is repeated, it matches the last one.